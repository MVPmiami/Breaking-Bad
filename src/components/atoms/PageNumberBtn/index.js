import React from "react";
import styles from "./style.module.scss";

const PageNumberBtn = ({ currentPage, numberPage, paginate }) => {
  return (
    <button
      className={`${styles.pageElement} ${
        currentPage === numberPage ? styles.active : ""
      }`}
      key={numberPage}
      onClick={() => {
        paginate(numberPage);
      }}
			data-test-pageNumBtn = {`pageNumBtn-${numberPage}`}
    >
      {numberPage}
    </button>
  );
};

export default PageNumberBtn;
