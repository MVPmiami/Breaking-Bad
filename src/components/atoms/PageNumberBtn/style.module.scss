@import "./../../../assets/styles/style.scss";

.pageElement {
  min-width: 2rem;
  min-height: 2rem;
  color: $dataText;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  &:hover {
    cursor: pointer;
    color: #f1b926;
  }
}
.pageElement.active {
  color: #f1b926;
  box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
  border-radius: 4px;
  background: #ffffff;
}
