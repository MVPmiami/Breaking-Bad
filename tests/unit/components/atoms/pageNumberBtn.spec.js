import React from "react";
import PageNumberBtn from "../../../../src/components/atoms/PageNumberBtn";

const setUp = (props) => shallow(<PageNumberBtn {...props} />);

describe("should render PageNumberBtn component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should contain .pageElement wrapper", () => {
    const wrapper = component.find(".pageElement");
    expect(wrapper.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("pageElement should contain .active class", () => {
    const wrapper = shallow(<PageNumberBtn currentPage={1} numberPage={2} />);
    wrapper.find(".active");
    expect(wrapper.length).toBe(1);
  });

  it("should call callback of pageElement", () => {
    const onBtnClick = jest.fn();
    const wrapper = shallow(<PageNumberBtn paginate={onBtnClick} />);
    wrapper.find(".pageElement").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });
});
