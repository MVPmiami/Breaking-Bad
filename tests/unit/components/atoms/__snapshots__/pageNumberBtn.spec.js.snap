// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PageNumberBtn component should contain .pageElement wrapper 1`] = `
<button
  className="pageElement active"
  onClick={[Function]}
/>
`;
